numpy/core/tests/test_numeric.py

```diff
@@ -2219,6 +2220,26 @@ def check_like_function(self, like_function, value, fill_value=False):
                 assert_equal(dz.dtype, np.dtype(dtype))
             self.compare_array_value(dz, value, fill_value)
 
+            # Test the 'shape' parameter
+            for s in self.shapes:
+                for o in 'CFA':
+                    sz = like_function(d, dtype=dtype, shape=s, order=o,
+                                       **fill_kwarg)
+                    assert_equal(sz.shape, s)
+                    if dtype is None:
+                        assert_equal(sz.dtype, d.dtype)
+                    else:
+                        assert_equal(sz.dtype, np.dtype(dtype))
+                    if o == 'C' or (o == 'A' and d.flags.c_contiguous):
+                        assert_(sz.flags.c_contiguous)
+                    elif o == 'F' or (o == 'A' and d.flags.f_contiguous):
+                        assert_(sz.flags.f_contiguous)
+                    self.compare_array_value(sz, value, fill_value)
+
+                if d.ndim != len(s):
+                    assert_raises(ValueError, like_function, d, dtype=dtype,
+                                  shape=s, order='K', **fill_kwarg)
```

Comment:
Needs an `else: assert_equal(np.argsort(like_function(...).strides) == np.argsory(d.strides)`

Reply:
I just added this condition. Note we have also to check for non-C/F-layouts while raising an exception.